### PR TITLE
Updated to use NuGet.Packaging

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@Eilon / @pranavkm

### CODE REVIEW

File: src/NuGetPackageVerifier/Rules/RequiredPackageMetadataRule.cs [hunk 1]
```diff
@@ -10,19 +11,20 @@ namespace NuGetPackageVerifier.Rules
...
     {
         public IEnumerable<PackageVerifierIssue> Validate(
-            IPackageRepository packageRepo,
-            IPackage package,
+            FileInfo nupkgFile,
+            IPackageMetadata package,
             IPackageVerifierLogger logger)
         {
-            if (string.IsNullOrEmpty(package.Summary))
+            var packageMetadata = package as IPackageMetadata;
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/RequiredPackageMetadataRule.cs]: Remove.

Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/RequiredPackageMetadataRule.cs]: :+1:

File: src/NuGetPackageVerifier/Program.cs [hunk 2]
```diff
@@ -91,14 +91,16 @@ public static int Main(string[] args)
             }.ToDictionary(t => t.GetType().Name, t => t);
 
             var nupkgsPath = args[0];
-            var localPackageRepo = new LocalPackageRepository(
-                new DefaultPackagePathResolver(nupkgsPath),
-                new CustomPhysicalFileSystem(nupkgsPath));
-            var numPackagesInRepo = localPackageRepo.GetPackages().Count();
+            var dirInfo = new DirectoryInfo(nupkgsPath);
+            var nupkgs = dirInfo.GetFiles("*.nupkg", SearchOption.TopDirectoryOnly);
+            var packages = nupkgs.ToDictionary<FileInfo, IPackageMetadata>(file => {
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Program.cs]: Brace on next line.

File: src/NuGetPackageVerifier/Program.cs [hunk 3]
```diff
@@ -91,14 +91,16 @@ public static int Main(string[] args)
...
+                var reader = new PackageArchiveReader(file.FullName);
+                return new PackageBuilder(reader.GetNuspec(), null);
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Program.cs]: What's the null param for?

Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Program.cs]: It is the basepath. Not necessary here. Will make it named param.

File: src/NuGetPackageVerifier/Rules/AssemblyHasAttributeRuleBase.cs [hunk 4]
```diff
@@ -14,13 +14,14 @@ namespace NuGetPackageVerifier.Rules
...
     {
         public IEnumerable<PackageVerifierIssue> Validate(
-            IPackageRepository packageRepo,
-            IPackage package,
+            FileInfo nupkgFile,
+            IPackageMetadata package,
             IPackageVerifierLogger logger)
         {
-            foreach (var currentFile in package.GetFiles())
+            var reader = new PackageArchiveReader(nupkgFile.FullName);
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/AssemblyHasAttributeRuleBase.cs]: Might need to Dispose this.

File: src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs [hunk 5]
```diff
@@ -33,38 +34,31 @@ public class AssemblyStrongNameRule : IPackageVerifierRule
 
                     try
                     {
-                        using (var packageFileStream = currentFile.GetStream())
+                        using (var packageFileStream = reader.GetStream(currentFile))
+                        using (var fileStream = new FileStream(assemblyPath, FileMode.Create))
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs]: `File.OpenWrite`?

File: NuGet.Config [hunk 6]
```diff
@@ -3,5 +3,6 @@
   <packageSources>
     <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
+    <add key="NuGetBuild" value="https://www.myget.org/F/nugetbuild/api/v3/index.json" />
```
Comment (pranavkm) [inline NuGet.Config]: We have a copy of their binaries in our feeds already. You probably don't need this. https://github.com/aspnet/CoreCLR/blob/dev/tools/MirrorLatestCoreCLR/Program.cs#L97

**Comment by pranavkm** [comment]
:shipit:

File: src/NuGetPackageVerifier/Program.cs [hunk 7]
```diff
@@ -91,14 +91,16 @@ public static int Main(string[] args)
             }.ToDictionary(t => t.GetType().Name, t => t);
 
             var nupkgsPath = args[0];
-            var localPackageRepo = new LocalPackageRepository(
-                new DefaultPackagePathResolver(nupkgsPath),
-                new CustomPhysicalFileSystem(nupkgsPath));
-            var numPackagesInRepo = localPackageRepo.GetPackages().Count();
+            var dirInfo = new DirectoryInfo(nupkgsPath);
+            var nupkgs = dirInfo.GetFiles("*.nupkg", SearchOption.TopDirectoryOnly);
```
Comment (Eilon) [inline src/NuGetPackageVerifier/Program.cs]: The old code had logic in it to ignore `.symbols.nupkg` files - do we still need that?

File: src/NuGetPackageVerifier/Rules/StrictSemanticVersionValidationRule.cs [hunk 8]
```diff
@@ -23,6 +25,7 @@ public class StrictSemanticVersionValidationRule : IPackageVerifierRule
             {
                 yield return PackageIssueFactory.NotSemanticVersion(package.Version);
             }
+            yield break;
```
Comment (Eilon) [inline src/NuGetPackageVerifier/Rules/StrictSemanticVersionValidationRule.cs]: Needed?

**Comment by Eilon** [comment]
Just some small comments, then :shipit:

**Pull request merged**